src/tools/illink/src/ILLink.Shared/DataFlow/ValueSet.cs

```diff
@@ -205,6 +207,24 @@ internal static ValueSet<TValue> Meet (ValueSet<TValue> left, ValueSet<TValue> r
 			return new ValueSet<TValue> (values);
 		}
 
+		internal static ValueSet<TValue> Intersection (ValueSet<TValue> left, ValueSet<TValue> right)
+		{
+			if (left._values == null)
+				return left;
+			if (right._values == null)
+				return right;
+
+			if (left._values is not EnumerableValues)
+				return right.Contains ((TValue) left._values) ? left.DeepCopy () : default;
+
+			if (right._values is not EnumerableValues)
+				return left.Contains ((TValue) right._values) ? right.DeepCopy () : default;
+
+			var values = new EnumerableValues (left.DeepCopy ());
+			values.IntersectWith (right.DeepCopy ());
```

Comment:
Do we need the DeepCopy here? I would expect the IntersectWith to not add the value fromt he parameter to the hashset - since it already has all the values it needs (unlike Union).